In [2]:
#Import Libraries

import pandas as pd
import numpy as np

In [4]:
#Load Dataset

df = pd.read_csv("D:/customer-churn-capstone/data/raw/customer_churn.csv")
df.head()

,CustomerID,Tenure,MonthlyCharges,TotalCharges,Contract,PaymentMethod,PaperlessBilling,SeniorCitizen,Churn
0,C00001,6,64,1540,One year,Credit Card,No,1,0
1,C00002,21,113,1753,Month-to-month,Electronic Check,Yes,1,0
2,C00003,27,31,1455,Two year,Credit Card,No,1,0
3,C00004,53,29,7150,Month-to-month,Electronic Check,No,1,0
4,C00005,16,185,1023,One year,Electronic Check,No,1,0


In [6]:
#Basic Dataset Overview

df.shape

(500, 9)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   CustomerID        500 non-null    object
 1   Tenure            500 non-null    int64 
 2   MonthlyCharges    500 non-null    int64 
 3   TotalCharges      500 non-null    int64 
 4   Contract          500 non-null    object
 5   PaymentMethod     500 non-null    object
 6   PaperlessBilling  500 non-null    object
 7   SeniorCitizen     500 non-null    int64 
 8   Churn             500 non-null    int64 
dtypes: int64(5), object(4)
memory usage: 35.3+ KB


In [8]:
df.isnull().sum()

CustomerID          0
Tenure              0
MonthlyCharges      0
TotalCharges        0
Contract            0
PaymentMethod       0
PaperlessBilling    0
SeniorCitizen       0
Churn               0
dtype: int64

In [9]:
df["TotalCharges"].dtype

dtype('int64')

In [10]:
#Drop Unnecessary Column

df = df.drop("CustomerID", axis=1)
df.head()

,Tenure,MonthlyCharges,TotalCharges,Contract,PaymentMethod,PaperlessBilling,SeniorCitizen,Churn
0,6,64,1540,One year,Credit Card,No,1,0
1,21,113,1753,Month-to-month,Electronic Check,Yes,1,0
2,27,31,1455,Two year,Credit Card,No,1,0
3,53,29,7150,Month-to-month,Electronic Check,No,1,0
4,16,185,1023,One year,Electronic Check,No,1,0


In [11]:
#Check Target Distribution

df["Churn"].value_counts()

Churn
0    447
1     53
Name: count, dtype: int64

Target Variable Analysis

The dataset is highly imbalanced:

447 customers (89.4%) did not churn

53 customers (10.6%) churned

This imbalance may bias the model toward predicting the majority class.
Therefore, evaluation metrics like Recall, F1-score, and ROC-AUC will be more important than Accuracy.

In [12]:
df.to_csv("../data/processed/churn_cleaned.csv", index=False)